src/Modules/UngradedModules.jl

```diff
@@ -491,6 +491,10 @@ function hash(a::AbstractFreeModElem, h::UInt)
   return hash(tuple(parent(a), coordinates(a)), h)
 end
 
+function Base.deepcopy_internal(a::AbstractFreeModElem, dict::IdDict)
```

Comment:
Is this the concrete type or a supertype?

Reply:
This is the abstract supertype of FreeModElem and FreeModElem_dec.

Reply:
There is the method https://github.com/oscar-system/Oscar.jl/blob/8f222b014ad54decbba1d581716d9ca99becf5d7/src/Modules/ModulesGraded.jl#L1325 detecting that the parent is decorated and then creating a decorated element.
This seems to be some kind of ugly hack, but all arithmetics, etc., for decorated free modules work exactly the same.

Reply:
Thanks for the explanation. I did not know we had something this unconventional.